proto/osmosis/cosmwasmpool/v1beta1/model/pool.proto

```diff
@@ -8,6 +8,21 @@ import "google/protobuf/timestamp.proto";
 
 option go_package = "github.com/osmosis-labs/osmosis/v19/x/cosmwasmpool/model";
 
+// CosmWasmPool represents the data serialized into state for each CW pool.
+//
+// Note: CW Pool has 2 pool models:
+// - CosmWasmPool which is a proto-generates store model used for serialization
+// into state.
+// - Pool struct that encapsulates the CosmWasmPool and wasmKeeper for calling
+// the contract.
+//
+// CosmWasmPool implements the poolmanager.PoolI interface but it panics on all
+// methods. The reason is that access to wasmKeeper is required to call the
+// contract.
+//
+// Instead, all interactions and poolmanager.PoolI methods are to be performed
+// on the Pool struct. The reason why we cannot have a Pool struct only is
+// because it cannot be serialized into state.
```

Comment:
Just out of curiosity, why not?

Reply:
I will elaborate in the comment.

The reason is that it has a wasmKeeper as field that is non-serializable.

Reply:
[f72cbc6](https://github.com/osmosis-labs/osmosis/pull/6484/commits/f72cbc6710dd2e4d6ebba3cd1ec29f32ab173139)